cdap-common/src/main/java/co/cask/cdap/common/utils/Networks.java

```diff
@@ -94,4 +96,15 @@ public static String normalizeWebappDiscoveryName(String name) throws Unsupporte
 
     return URLEncoder.encode(name, Charsets.UTF_8.name());
   }
+
+
+  /**
+   * Get the filename part from a {@link URI}. For a URI "file:///tmp/foo.jar", "foo.jar" will be returned.
+   *
+   * @param uri The {@link URI} to get filename from.
+   * @return filename.
+   */
+  public static String getURIFilename(URI uri) {
+    return new File(uri.getPath()).getName();
```

Comment:
Since the only usage of this method is actually inside `LocalizationUtils.getLocalizedName()`, we can just do the logic there (meaning removing this). BTW, the logic is not right. It shouldn't use `File` to get the name from URI. You should do:

```java
String path = URI.getPath();
int idx = path.lastIndexOf("/");
return idx >= 0 ? path.substring(idx + 1) : path;
```